private/bufpkg/bufmodule/module_data.go

```diff
@@ -140,24 +147,15 @@ func newModuleData(
 				}
 			case DigestTypeB5:
 				// Call unexported func instead of exported method to avoid deadlocking on checking the digest again.
-				declaredDepModuleKeys, err := moduleData.getDeclaredDepModuleKeys()
+				depModuleKeys, err := moduleData.getDepModuleKeys()
 				if err != nil {
 					return err
 				}
-				// This isn't the Digest as computed by the Module exactly, as the Module uses
-				// file imports to determine what the dependencies are. However, this is checking whether
-				// or not the digest of the returned information matches the digest we expected, which is
-				// what we need for this use case (tamper-proofing). What we are looking for is "does the
-				// digest from the ModuleKey match the files and dependencies returned from the remote
-				// provider of the ModuleData?" The mismatch case is that a file import changed/was removed,
-				// which may result in a different computed set of dependencies, but in this case, the
-				// actual files would have changed, which will result in a mismatched digest anyways, and
-				// tamper-proofing failing.
-				//
-				// This mismatch is a bit weird, however, and also results in us effectively computing
-				// the digest twice for any remote module: once here, and once within Module.Digest,
-				// which does have a slight performance hit.
-				actualDigest, err = getB5DigestForBucketAndDepModuleKeys(ctx, bucket, declaredDepModuleKeys)
+				// The B5 digest is calculated based on the dependencies.
+				// Dependencies are not required to be resolved to a Module to calculate the digest.
+				// Each dependent ModuleKey is a reference to a specific commit of a module
+				// that includes the dependencies expected Digest.
```

Comment:
This comment was a little hard for me to parse. Maybe something like this?
> The B5 digest is calculated based on the dependencies. We do not need to construct a Module to calculate a digest as we can trust that `DepModuleKeys` is the complete set of deps for the Module and is exactly the deps that were present in the ModuleSet when this Module was pushed.

Reply:
I'm not sure trust is correct, as it is always validated if used. What about:
> // The B5 digest is calculated based on the dependencies.
> // The dependencies are not required to be resolved to a Module to calculate this Digest.
> // Each ModuleKey includes the expected Digest which is validated when loading that dependencies ModuleData, if needed.